web/ui/src/Services/ServiceDetailsController.js

```diff
@@ -947,8 +955,58 @@
                     return $scope.indent(depth - offset);
                 };
 
-                $scope.setCurrentTreeState = function () {
-                    $scope.serviceTreeState[$scope.currentService.id] = {};
+                $scope.restoreTreeState = function () {
+
+                    if (!$scope.currentService) {
+                        return;
+                    }
+                    // initialize at first call
+                    if (!$scope.serviceTreeState[$scope.currentService.id]) {
+                        $scope.serviceTreeState[$scope.currentService.id] = {};
+                    }
+                    // delete hidden services state upon re-entry
+                    let treeState = $scope.serviceTreeState[$scope.currentService.id];
+                    Object.keys(treeState).forEach(k => {
+                        if (treeState[k].hidden) {
+                            delete treeState[k];
+                        }
+                    });
+                    $scope.serviceTreeState[$scope.currentService.id] = treeState;
+                    
+                    $scope.updateTreeState();
+                    $scope.currentTreeState = treeState;
+
+                };
+
+                $scope.updateTreeState = function () {        
+                    let treeState = $scope.serviceTreeState[$scope.currentService.id];
+                    $scope.currentService.subservices.forEach(topLvlSvc => {
```

Comment:
You should be able to get rid of the `recurse` IIFE below by changing this line to 

    $scope.currentService.subservices.forEach(function recurse(service) {

and deleting the old `recurse` stuff at line 985 and 1007